gpt4all-chat/database.cpp

```diff
@@ -1038,6 +1038,53 @@ void Database::handleDocumentError(const QString &errorMessage,
     qWarning() << errorMessage << document_id << document_path << error;
 }
 
+static QString recursiveTextSplitter(QString &text, int &splitLevel, size_t maxChars,
+    const QList<QString> &splitters, int splitterIndex = 0)
```

Comment:
This seems wrong. splitLevel is a reference to a local variable in the caller that is never actually used. It is set by all levels of recursion, but not in any clear order. So the return statement that happens last, whichever level that may be, will set it.

If you do keep that variable, this would make more sense to me if you used multiple return values, so the value from a recursive call would be dealt with explicitly. In C++17 this looks like:
```c++
std::pair<QString, int> foo() {
    // ...
    return {chunk, level};
}

void bar() {
    auto [chunk, level] = foo();
}
```

Reply:
I'm combining them into one. No reason to have both.